### ISSUE TITLE
Display Subcommands in Help Menu

### ISSUE THREAD

**Issue by joel-olazagasti** [kangalioo/poise#148]
I can't quite figure out a way to populate subcommands/groups into the built-in help menu. I've defaulted to manually listing the subcommands in the help text, but that obviously requires manual upkeep. Maybe there's a way to use `help_text_fn` to accomplish this? Would greatly appreciate if anyone has some direction here (or telling me it's not possible with the current implementation)

**Comment by kangalio** [comment]
Indeed, the built-in help menu only prints top-level commands. But support could be added behind a new `HelpConfiguration::include_subcommands: bool` flag. Not sure how the rendered text should look like. Suggestions welcome

In the meantime, you could make your own help command. Commands are accessible via `ctx.framework().options().commands`. Example:

```rust
let mut text = String::new();
for command in &ctx.framework().options().commands {
    text += &format!("{}: {}\n", command.name, command.description);
    for command in &command.subcommands {
        text += &format!("    {}: {}\n", command.name, command.description);
    }
}
ctx.say(text).await?;
```

**Comment by joel-olazagasti** [comment]
Appreciate the help! I'll play around with some formatting/optionality and see what works well and report back.

**Comment by kangalio** [comment]
Implemented in ec869bbd7cc97a9d9e8e0d83c239802b142b464f
